src/test/java/org/mule/test/extension/file/common/AbstractFileAttributesTestCase.java

```diff
@@ -135,13 +135,117 @@ public void bothConstructorAssignEqualFileNamesForParentDirectoryWithTrailingSla
     assertFileName(path);
   }
 
+  @Test
+  public void bothConstructorAssignAnEmptyBasePath() {
+    ConcreteFileAttributes uriAttributes = new ConcreteFileAttributes(createUri("", TEST_FILENAME));
+    assertThat(uriAttributes.getPath(), equalTo(TEST_FILENAME));
+  }
+
+  @Test
+  public void bothConstructorAssignAnEmptyFilePath() {
+    path = "/test";
+    ConcreteFileAttributes uriAttributes = new ConcreteFileAttributes(createUri(path, ""));
+    assertThat(uriAttributes.getPath(), equalTo(path));
+  }
+
+  @Test
+  public void bothConstructorAssignAnEmptyFilePathWithoutDash() {
+    ConcreteFileAttributes uriAttributes = new ConcreteFileAttributes(createUri(TEST_FILENAME, ""));
+    assertThat(uriAttributes.getPath(), equalTo(TEST_FILENAME));
+  }
+
   private void assertFileName(String path) {
     ConcreteFileAttributes pathAttributes = new ConcreteFileAttributes(Paths.get(path));
     ConcreteFileAttributes uriAttributes = new ConcreteFileAttributes(createUri(path));
 
     assertThat(pathAttributes.getName(), equalTo(uriAttributes.getName()));
   }
 
+  @Test
+  public void bothConstructorWithBasePathAssignInSpecificFolderWithSeparator2() {
+    ConcreteFileAttributes pathAttributes = new ConcreteFileAttributes(Paths.get("/myFile.txt"));
+    // Basepath parameter will be ignored when tne filepath is absolute (when it start with slash)
+    ConcreteFileAttributes uriAttributes = new ConcreteFileAttributes(createUri("/root/", "/myFile.txt"));
```

Comment:
in this case I would expect the result as "/root/myfile.txt", the condition to check if the path is absolute checks only that the filePath started with a "/" but that doesn't mean that the path is absolute right? in Unix I would expect a "~/" to determine that the path is absolute and in Windows I would expect a "X:\"